src/SOS/Strike/platform/runtimeimpl.cpp

```diff
@@ -496,41 +530,107 @@ HRESULT Runtime::GetCorDebugInterface(ICorDebugProcess** ppCorDebugProcess)
         skuId = CLR_ID_V4_DESKTOP;
     }
 #endif
-    CLRDebuggingImpl* pDebuggingImpl = new CLRDebuggingImpl(skuId, IsWindowsTarget());
-    hr = pDebuggingImpl->QueryInterface(IID_ICLRDebugging, (LPVOID *)&pClrDebugging);
-    if (FAILED(hr))
+    const char* dacFilePath = GetDacFilePath();
+    if (dacFilePath == nullptr)
+    {
+        ExtErr("Could not find matching DAC\n");
+        return CORDBG_E_NO_IMAGE_AVAILABLE;
+    }
+    ArrayHolder<WCHAR> pDacModulePath = new WCHAR[MAX_LONGPATH + 1];
+    int length = MultiByteToWideChar(CP_ACP, 0, dacFilePath, -1, pDacModulePath, MAX_LONGPATH);
+    if (0 >= length)
     {
-        delete pDebuggingImpl;
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        ExtErr("MultiByteToWideChar() DAC FAILED %08x\n", hr);
         return hr;
     }
-
-    ToRelease<ICorDebugMutableDataTarget> pCorDebugDataTarget = new CorDebugDataTarget;
-    pCorDebugDataTarget->AddRef();
-
-    ToRelease<ICLRDebuggingLibraryProvider> pCorDebugLibraryProvider = new CorDebugLibraryProvider(this);
-    pCorDebugLibraryProvider->AddRef();
-
-    CLR_DEBUGGING_VERSION clrDebuggingVersionRequested = {0};
-    clrDebuggingVersionRequested.wMajor = 4;
-
-    CLR_DEBUGGING_VERSION clrDebuggingVersionActual = {0};
-
+    const char* dbiFilePath = GetDbiFilePath();
+    if (dbiFilePath == nullptr) 
+    {
+        ExtErr("Could not find matching DBI\n");
+        return CORDBG_E_NO_IMAGE_AVAILABLE;
+    }
+    HMODULE hDbi = LoadLibraryA(dbiFilePath);
+    if (hDbi == NULL)
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        ExtErr("LoadLibraryA(%s) FAILED %08x\n", dbiFilePath, hr);
+        return hr;
+    }
+    CLR_DEBUGGING_VERSION clrDebuggingVersionRequested = {0, 4, 0, 0, 0};
     CLR_DEBUGGING_PROCESS_FLAGS clrDebuggingFlags = (CLR_DEBUGGING_PROCESS_FLAGS)0;
+    ToRelease<ICorDebugMutableDataTarget> pDataTarget = new CorDebugDataTarget;
```

Comment:
Can we make all this get function and validate a helper? This feels like a really long function to read now.